@here/harp-features-datasource/lib/FeaturesDataSource.ts

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+    Feature,
+    FeatureCollection,
+    FeatureGeometry,
+    LineString,
+    MultiLineString,
+    MultiPoint,
+    MultiPolygon,
+    Point,
+    Polygon
+} from "@here/harp-datasource-protocol";
+import { GeoJsonDataProvider } from "@here/harp-geojson-datasource";
+import { OmvDataSource } from "@here/harp-omv-datasource";
+import {
+    MapViewFeature,
+    MapViewLineFeature,
+    MapViewMultiLineFeature,
+    MapViewMultiPointFeature,
+    MapViewMultiPolygonFeature,
+    MapViewPointFeature,
+    MapViewPolygonFeature
+} from "./Features";
+
+const NAME = "user-features-datasource";
+
+const DEFAULT_GEOJSON: FeatureCollection = {
+    type: "FeatureCollection",
+    features: []
+};
+
+/**
+ * [[DataSource]] implementation to use for the addition of custom features.
+ */
+export class FeaturesDataSource extends OmvDataSource {
+    private m_featureCollection: FeatureCollection = {
+        type: "FeatureCollection",
+        features: []
+    };
+
+    /**
+     * Builds a `FeaturesDataSource`.
+     *
+     * @param workerTilerUrl Worker tiler URL. Defaults to `./decoder.bundle.ts` in the
+     * [[ConcurrentTilerFacade]].
+     */
+    constructor(workerTilerUrl?: string) {
+        super({
+            dataProvider: new GeoJsonDataProvider(NAME, DEFAULT_GEOJSON, workerTilerUrl)
+        });
+    }
+
+    /**
+     * Creates a polygon feature.
+     *
+     * @param coordinates The geometry of the feature.
+     * @param style The style to draw the feature with.
+     */
+    createPolygon(coordinates: Polygon["coordinates"], properties?: {}): MapViewPolygonFeature {
```

Comment:
these create* methods don't need to be part of this DataSource do they? I think it would be clearer to have them as free functions separate from the data source